pages/api/roles/index.ts

```diff
@@ -1,5 +1,129 @@
+import { isEmpty } from '@firebase/util';
+import { PrismaClient, RoleType } from '@prisma/client';
 import type { NextApiRequest, NextApiResponse } from 'next';
+import { ApiResponse, EmptyPayload, StatusMessage, StatusMessageType } from '../../../types/apiResponse';
+import { RoleData, RoleListData, RolePostData } from '../../../types/role';
+import { withAuthUser } from '../../../utils/auth/jwtHelpers';
+import {
+  createJsonResponse,
+  HTTP_GET_METHOD,
+  HTTP_POST_METHOD,
+  HTTP_STATUS_BAD_REQUEST,
+  HTTP_STATUS_CREATED,
+  HTTP_STATUS_OK,
+  rejectHttpMethod,
+} from '../../../utils/http/httpHelper';
+import { createIfPossible } from '../../../utils/prisma/prismaHelpers';
 
-export default function handler(req: NextApiRequest, res: NextApiResponse) {
-  res.status(200).end(`Request sent to ${req.url}\n`);
+const prisma = new PrismaClient();
+
+enum MessageType {
+  INVALID_COMPANY_ID,
+  COMPANY_DOES_NOT_EXISTS,
+  EMPTY_TITLE,
+  INVALID_TYPE,
+  INVALID_YEAR,
+  ROLE_CREATED_SUCCESSFULLY,
+}
+
+const messages = new Map<MessageType, StatusMessage[]>([
+  [
+    MessageType.INVALID_COMPANY_ID,
+    [{ type: StatusMessageType.Error, message: 'The company for this role is invalid.' }],
+  ],
+  [
+    MessageType.COMPANY_DOES_NOT_EXISTS,
+    [{ type: StatusMessageType.Error, message: 'The company for this role does not exists.' }],
+  ],
+  [MessageType.EMPTY_TITLE, [{ type: StatusMessageType.Error, message: 'Role title is empty.' }]],
+  [MessageType.INVALID_TYPE, [{ type: StatusMessageType.Error, message: 'Role type is invalid.' }]],
+  [MessageType.INVALID_YEAR, [{ type: StatusMessageType.Error, message: 'Role year is invalid.' }]],
```

Comment:
Could we provide more descriptive status messages? For a role year to be invalid could mean many things other than not being a `number`